src/coreComponents/physicsSolvers/multiphysics/MultiResolutionHFSolver.hpp

```diff
@@ -0,0 +1,150 @@
+   * @return string that contains the catalog name to generate a new NodeManager object through the object catalog.
+   */
+  static string catalogName()
+  {
+    return "MultiResolutionHF";
+  }
+
+  virtual void RegisterDataOnMesh( Group & MeshBodies );
+
+  virtual void
+  implicitStepSetup( real64 const & time_n,
+                     real64 const & dt,
+                     DomainPartition & domain ) override final;
+
+  virtual void resetStateToBeginningOfStep( DomainPartition & domain ) override;
+
+  virtual real64 solverStep( real64 const & time_n,
+                             real64 const & dt,
+                             int const cycleNumber,
+                             DomainPartition & domain ) override;
+
+  //virtual void updateNodeMaps( MeshLevel const & base, MeshLevel const & patch );
+
+  virtual void setInitialCrackDamageBCs( DofManager const & dofManager,
+                                         CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                                         MeshLevel const & patch,
+                                         MeshLevel const & base );
+
+  virtual void findPhaseFieldTip( R1Tensor & tip,
+                                  MeshLevel const & patch );
+
+  virtual void prepareSubProblemBCs( MeshLevel const & base,
+                                     MeshLevel & patch );
+
+  real64 splitOperatorStep( real64 const & time_n,
+                            real64 const & dt,
+                            integer const cycleNumber,
+                            DomainPartition & domain );
+
+  // virtual void setNextDt( real64 const & currentDt,
+  //                         real64 & nextDt ) override;
+
+
+  struct viewKeyStruct : SolverBase::viewKeyStruct
+  {
+    constexpr static char const * baseSolverNameString() { return "baseSolverName"; }
+
+    constexpr static char const * patchSolverNameString() { return "patchSolverName"; }
+
+    constexpr static char const * contactRelationNameString() { return "contactRelationName"; }
+
+    constexpr static char const * surfaceGeneratorNameString() { return "surfaceGeneratorName"; }
+
+    constexpr static char const * maxNumResolvesString() { return "maxNumResolves"; }
+
+    constexpr static char const * initialBaseTipString() { return "initialBaseTip"; }
+
+    constexpr static char const * initialTipElementIndexString() { return "initialTipElementIndex"; }
+
+  };
+
+protected:
+  virtual void postProcessInput() override final;
+
+  virtual void
+  initializePostInitialConditionsPreSubGroups() override final;
+
+private:
+
+  // name of the base efem solver
+  string m_baseSolverName;
+
+  //name of patch phase-field fracture solver
+  string m_patchSolverName;
+
+  // list of damage dofs to be fixed in the subdomain boundary
+  array1d< localIndex > m_nodeFixDamage;
+
+  // list of disp dofs to be fixed in the subdomain boundary
+  array1d< localIndex > m_nodeFixDisp;
```

Comment:
I am not sure you are using the right type of containers for these. Why don't you use sets (`sortedArray`) instead of arrays? I think it would make sense since you are always checking if they are unique. Maybe they can even jsut be std objects?